pkg/kafka/partition/reader.go

```diff
@@ -114,7 +154,251 @@ func (p *Reader) run(ctx context.Context) error {
 	return nil
 }
 
-func (p *Reader) startFetchLoop(ctx context.Context) <-chan []Record {
+func (p *Reader) fetchLastCommittedOffset(ctx context.Context) int64 {
+	// We manually create a request so that we can request the offset for a single partition
+	// only, which is more performant than requesting the offsets for all partitions.
+	req := kmsg.NewPtrOffsetFetchRequest()
+	req.Topics = []kmsg.OffsetFetchRequestTopic{{Topic: p.kafkaCfg.Topic, Partitions: []int32{p.partitionID}}}
+	req.Group = p.consumerGroup
+
+	resps := p.client.RequestSharded(ctx, req)
+
+	// Since we issued a request for only 1 partition, we expect exactly 1 response.
+	if expected := 1; len(resps) != expected {
```

Comment:
What's the point of the expected variable when it's not used outside the condition?

```suggestion
	if len(resps) != 1 {
```